static/app/views/settings/components/dataSecrecy/index.tsx

```diff
@@ -0,0 +1,170 @@
+
+  const {data, refetch} = useApiQuery<WaiverData>(
+    [`/organizations/${organization.slug}/data-secrecy/`],
+    {
+      staleTime: 3000,
+      retry: (failureCount, error) => failureCount < 3 && error.status !== 404,
+    }
+  );
+
+  const hasValidTempAccess =
+    allowDate?.access_end && moment().toISOString() < allowDate.access_end;
+
+  useEffect(() => {
+    if (data?.access_end) {
+      setAllowDate(data);
+      // slice it to yyyy-MM-ddThh:mm format (be aware of the timezone)
+      const localDate = moment(data.access_end).local();
+      setAllowDateFormData(localDate.format('YYYY-MM-DDTHH:mm'));
+    }
+  }, [data]);
+
+  const updateAllowedAccess = async (value: boolean) => {
+    try {
+      await api.requestPromise(`/organizations/${organization.slug}/`, {
+        method: 'PUT',
+        data: {allowSuperuserAccess: value},
+      });
+      setAllowAccess(value);
+      addSuccessMessage(t('Successfully updated access.'));
+    } catch (error) {
+      addErrorMessage(t('Unable to save changes.'));
+    }
+  };
+
+  const updateTempAccessDate = async () => {
+    if (!allowDateFormData) {
+      try {
+        await api.requestPromise(`/organizations/${organization.slug}/data-secrecy/`, {
+          method: 'DELETE',
+        });
+        setAllowDate({access_start: '', access_end: ''});
+        addSuccessMessage(t('Successfully removed temporary access window.'));
+      } catch (error) {
+        addErrorMessage(t('Unable to remove temporary access window.'));
+      }
+
+      return;
+    }
+
+    // maintain the standard format of storing the date in UTC
+    // even though the api should be able to handle the local time
+    const nextData = {
+      access_start: moment().utc().toISOString(),
+      access_end: moment.tz(allowDateFormData, moment.tz.guess()).utc().toISOString(),
+    };
+
+    try {
+      await await api.requestPromise(
+        `/organizations/${organization.slug}/data-secrecy/`,
+        {
+          method: 'PUT',
+          data: nextData,
+        }
+      );
+      setAllowDate(nextData);
+      addSuccessMessage(t('Successfully updated temporary access window.'));
+    } catch (error) {
+      addErrorMessage(t('Unable to save changes.'));
+      setAllowDateFormData('');
+    }
+    // refetch to get the latest waiver data
+    refetch();
+  };
+
+  const allowAccessProps: BooleanFieldProps = {
+    name: 'allowSuperuserAccess',
+    label: t('Allow access to Sentry employees'),
+    help: t(
+      'Sentry employees will not have access to your organization unless granted permission'
+    ),
+    'aria-label': t(
+      'Sentry employees will not have access to your data unless granted permission'
+    ),
+    value: allowAccess,
+    disabled: organization.access.includes('org:write'),
+    onBlur: updateAllowedAccess,
+  };
+
+  const allowTempAccessProps: DateTimeFieldProps = {
+    name: 'allowTemporarySuperuserAccess',
+    label: t('Allow temporary access for Sentry employees'),
```

Comment:
nit: My english is not good so I don't know which one is wrong but one of them has to be `Allow temporary access for Sentry employees` vs `Allow access to Sentry employees`. Is it to? or is it for? 😛

Reply:
got opinion from docs and am switching to "to" lol